### ISSUE TITLE
venmic.log file gets created in my home folder

### ISSUE THREAD

**Issue by notpeelz** [Vencord/Vesktop#234]
`~/venmic.log`:
```
[2023-11-11 15:36:26.201] [venmic] [debug] [has_pipewire] pulse-server was pulseaudio (on pipewire 0.3.84)
[2023-11-11 15:36:26.213] [venmic] [info] found speakers: 40
```

### System information
```
$ . /etc/os-release; echo "$NAME"
Arch Linux
$ uname -srvom
Linux 6.6.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 08 Nov 2023 16:05:38 +0000 x86_64 GNU/Linux
$ pacman -Qi vencord-desktop-git | head -n2
Name            : vencord-desktop-git
Version         : r221.a232af0-1
```

**Comment by Vendicated** [comment]
can't repro. please provide more info. what did you do exactly that led to this?

**Comment by notpeelz** [comment]
1. launch vencord-desktop
    ```
    mkdir ~/test
    cd ~/test
    gtk-launch vencord-desktop
   ```
2. join a vc and screen share
3. see `~/test/venmic.log` get created

**Comment by Curve** [comment]
Venmic keeps a log file since latest release and will create this file in the directory it was launched from

**Comment by notpeelz** [comment]
I'm launching vencord-desktop via my DE/compositor, whose working directory defaults to the `$HOME` folder in my case. According to the [Freedesktop XDG spec](https://specifications.freedesktop.org/basedir-spec/0.8/ar01s03.html), log files should be stored in `$XDG_STATE_HOME` (`~/.local/state` by default).

Maybe the log file could also be made optional / enabled via an environment variable?

**Comment by D3SOX** [comment]
FYI, this also currently breaks streaming with audio when it's started from a directory without write permissions

**Comment by Curve** [comment]
Fixed in venmic `v2.1.1`.

The log file will now be placed in either:

* $XDG_STATE_HOME
* $HOME/.local/state _(In case state home was not set)_
* A temporary directory

The logging is now also off by default and will only be enable when `VENMIC_ENABLE_LOG` is set

**Comment by Curve** [comment]
Will submit a PR to update here once [this bad boy finishes](https://github.com/Vencord/venmic/actions/runs/6836970783/job/18592444935)
